adapters/python/sinks/kafka_redis.py

```diff
@@ -215,30 +224,81 @@ async def store_frame_content(
 
         content_key = f'{self._config.redis.key_prefix}:{frame.uuid}'
         location = f'{self._config.redis.host}:{self._config.redis.port}:{self._config.redis.db}/{content_key}'
+
+        if await self.need_to_store_content_to_redis(frame, content_key, location):
+            self._logger.debug(
+                'Storing content of the frame %s from source %s to Redis location %r (%s bytes)',
+                frame.pts,
+                frame.source_id,
+                location,
+                len(content),
+            )
+            await self._redis_client.set(
+                content_key,
+                content,
+                ex=self._config.redis.ttl_seconds,
+            )
+
+        return VideoFrameContent.external(ExternalFrameType.REDIS.value, location)
+
+    async def need_to_store_content_to_redis(
+        self,
+        frame: VideoFrame,
+        content_key: str,
+        location: str,
+    ):
+        """Check if the frame content needs to be stored to Redis.
+
+        The frame content doesn't need tobe stored in Redis when the following conditions are met:
```

Comment:
"The frame content doesn't need tobe..." => "The frame content doesn't need to be..."